requirements.txt

```diff
@@ -1,12 +1,16 @@
-pandas
+deprecation
 geopandas>=0.9
+libpysal
+mapclassify
 matplotlib
-scikit-learn>=0.21.3
-seaborn
 numpy
+pandana
```

Comment:
nice, thanks! though a few of these are optional deps so we need them in requirements_tests.txt to ensure full coverage but leave them out of the general requirements to avoid forcing them to install with meta.

Reply:
OK. Which are optional that I should put in `requirements_tests.txt`?

Reply:
i think its 

ipywidgets
pandana
urbanaccess
quilt3

though im not sure if we actually use ipywidgets anywhere

Reply:
or twine

Reply:
the new notebooks in   #174 require some dependencies as well

- watermark
- ipywidgets (IProgress)

Reply:
if we're going to test the notebooks in the CI, then we'd need to add geosnap as well

Reply:
Could we move those deps into requirements_doc? 

It can be a pain point for a user to try to run the demo notebooks and hit the tracebacks.

Reply:
OK, I'll add `watermark` and `ipywidgets ` into `requirements_doc.txt`. Also, there is now [a solution](https://github.com/pysal/spaghetti/pull/644/files#diff-85933aa74a2d66c3e4dcdf7a9ad8397f5a7971080d34ef1108296a7c6b69e7e3R37) for the `sphinxcontrib-bibtex` issue and we can remove [the pin](https://github.com/pysal/segregation/blob/403bcf6db748c366d9083ffc98cf66dfa1f536a4/.ci/39.yml#L27) in `.ci/39.yml`.

Reply:
so i think watermark, ipywidgets, etc need to be in requirements_tests not requirements_doc (we need them installed to run tests, not to build the docs though)